src/socket.rs

```diff
@@ -59,6 +60,7 @@ pub enum SocketError {
 /// related to channels is exposed there.
 pub struct Socket {
     url: Arc<Url>,
+    status: ArcStatus,
```

Comment:
This could also be state: ArcSwap<State>

Reply:
Since the `socket::Listener` `State` is an internal implementation detail and the Socket just needs to know which State it is in, but not the internals of that state, I want to get the `State` out of `Socket`.